### PR TITLE
twitter roberta

### PR DESCRIPTION
**Pull request by liangzixuan**
python fine-tuning-llm.py
[2025-03-05 01:18:17,522] INFO __main__:<module>:308 - Starting...
[2025-03-05 01:18:17,522] INFO __main__:step_04_predict_sentiment:270 - Using model: base...
[2025-03-05 01:18:17,872] INFO __main__:predict_sentiment:247 - Predict: [I loved this movie!                                                                                 ] :        Sentiment: positive, Score: 0.99. Actual sentiment: Positive
[2025-03-05 01:18:17,926] INFO __main__:predict_sentiment:247 - Predict: [This was the worst experience ever.                                                                 ] :        Sentiment: negative, Score: 0.94. Actual sentiment: Negative
[2025-03-05 01:18:17,979] INFO __main__:predict_sentiment:247 - Predict: [It was okay, but not great.                                                                         ] :        Sentiment: negative, Score: 0.78. Actual sentiment: Neutral
[2025-03-05 01:18:18,032] INFO __main__:predict_sentiment:247 - Predict: [This was the worst movie I've ever seen. A complete waste of time.                                  ] :        Sentiment: negative, Score: 0.96. Actual sentiment: negative
[2025-03-05 01:18:18,084] INFO __main__:predict_sentiment:247 - Predict: [Absolutely awful! I regret watching this movie.                                                     ] :        Sentiment: negative, Score: 0.95. Actual sentiment: negative
[2025-03-05 01:18:18,140] INFO __main__:predict_sentiment:247 - Predict: [The plot was weak, and the acting was even worse.                                                   ] :        Sentiment: negative, Score: 0.95. Actual sentiment: negative
[2025-03-05 01:18:18,190] INFO __main__:predict_sentiment:247 - Predict: [It had a few good moments, but overall, it was disappointing.                                       ] :        Sentiment: negative, Score: 0.85. Actual sentiment: negative
[2025-03-05 01:18:18,245] INFO __main__:predict_sentiment:247 - Predict: [Not bad, but I wouldn’t watch it again.                                                             ] :        Sentiment: negative, Score: 0.67. Actual sentiment: negative
[2025-03-05 01:18:18,298] INFO __main__:predict_sentiment:247 - Predict: [Es war der schlimmste Film, den ich je gesehen habe.                                                ] :        Sentiment: neutral, Score: 0.70. Actual sentiment: negative
[2025-03-05 01:18:18,345] INFO __main__:predict_sentiment:247 - Predict: [Der Film war eine große Enttäuschung.                                                               ] :        Sentiment: neutral, Score: 0.73. Actual sentiment: negative
[2025-03-05 01:18:18,357] INFO __main__:predict_sentiment:247 - Predict: [Ein durchschnittlicher Film mit einigen guten Momenten.                                             ] :        Sentiment: neutral, Score: 0.80. Actual sentiment: neutral
[2025-03-05 01:18:18,368] INFO __main__:predict_sentiment:247 - Predict: [Der Film hatte eine interessante Handlung.                                                          ] :        Sentiment: neutral, Score: 0.73. Actual sentiment: positive
[2025-03-05 01:18:18,379] INFO __main__:predict_sentiment:247 - Predict: [Fantastischer Film! Ich habe jede Minute genossen.                                                  ] :        Sentiment: positive, Score: 0.97. Actual sentiment: positive
[2025-03-05 01:18:18,428] INFO __main__:predict_sentiment:247 - Predict: [Questo film era terribile, non lo consiglio a nessuno.                                              ] :        Sentiment: neutral, Score: 0.79. Actual sentiment: negative
[2025-03-05 01:18:18,441] INFO __main__:predict_sentiment:247 - Predict: [Non mi è piaciuto, la trama era confusa.                                                            ] :        Sentiment: neutral, Score: 0.84. Actual sentiment: negative
[2025-03-05 01:18:18,452] INFO __main__:predict_sentiment:247 - Predict: [Un film discreto, ma niente di speciale.                                                            ] :        Sentiment: neutral, Score: 0.86. Actual sentiment: neutral
[2025-03-05 01:18:18,462] INFO __main__:predict_sentiment:247 - Predict: [Molto bello, una storia ben raccontata.                                                             ] :        Sentiment: neutral, Score: 0.84. Actual sentiment: positive
[2025-03-05 01:18:18,513] INFO __main__:predict_sentiment:247 - Predict: [Capolavoro assoluto! Da vedere e rivedere.                                                          ] :        Sentiment: neutral, Score: 0.76. Actual sentiment: positive
[2025-03-05 01:18:18,525] INFO __main__:predict_sentiment:247 - Predict: [Ce film était terrible, je ne le recommande pas.                                                    ] :        Sentiment: negative, Score: 0.93. Actual sentiment: negative
[2025-03-05 01:18:18,537] INFO __main__:predict_sentiment:247 - Predict: [Un film ennuyeux avec peu de bons moments.                                                          ] :        Sentiment: neutral, Score: 0.89. Actual sentiment: negative
[2025-03-05 01:18:18,555] INFO __main__:predict_sentiment:247 - Predict: [Un film moyen, mais agréable.                                                                       ] :        Sentiment: neutral, Score: 0.88. Actual sentiment: neutral
[2025-03-05 01:18:18,566] INFO __main__:predict_sentiment:247 - Predict: [J'ai aimé l'histoire, c'était intéressant.                                                          ] :        Sentiment: neutral, Score: 0.77. Actual sentiment: positive
[2025-03-05 01:18:18,576] INFO __main__:predict_sentiment:247 - Predict: [Incroyable! Un des meilleurs films que j'ai vus.                                                    ] :        Sentiment: positive, Score: 0.92. Actual sentiment: positive
[2025-03-05 01:18:18,587] INFO __main__:predict_sentiment:247 - Predict: [Esta película fue horrible, no la recomiendo.                                                       ] :        Sentiment: negative, Score: 0.93. Actual sentiment: negative
[2025-03-05 01:18:18,638] INFO __main__:predict_sentiment:247 - Predict: [Un poco aburrida, pero tuvo algunos momentos buenos.                                                ] :        Sentiment: neutral, Score: 0.85. Actual sentiment: negative
[2025-03-05 01:18:18,650] INFO __main__:predict_sentiment:247 - Predict: [Fue un buen filme, pero nada espectacular.                                                          ] :        Sentiment: neutral, Score: 0.74. Actual sentiment: neutral
[2025-03-05 01:18:18,661] INFO __main__:predict_sentiment:247 - Predict: [Gran película, realmente disfruté la trama.                                                         ] :        Sentiment: neutral, Score: 0.81. Actual sentiment: positive
[2025-03-05 01:18:18,671] INFO __main__:predict_sentiment:247 - Predict: [Maravilloso! Lo volvería a ver muchas veces.                                                        ] :        Sentiment: positive, Score: 0.65. Actual sentiment: positive
[2025-03-05 01:18:18,672] INFO __main__:step_04_predict_sentiment:304 - Accuracy on sample_sentences_2.json: 50.00%

[2025-03-05 01:18:18,673] INFO __main__:step_04_predict_sentiment:270 - Using model: fine_tuned...
[2025-03-05 01:18:18,854] INFO __main__:predict_sentiment:247 - Predict: [I loved this movie!                                                                                 ] :        Sentiment: neutral, Score: 0.99. Actual sentiment: Positive
[2025-03-05 01:18:18,868] INFO __main__:predict_sentiment:247 - Predict: [This was the worst experience ever.                                                                 ] :        Sentiment: negative, Score: 1.00. Actual sentiment: Negative
[2025-03-05 01:18:18,884] INFO __main__:predict_sentiment:247 - Predict: [It was okay, but not great.                                                                         ] :        Sentiment: negative, Score: 0.91. Actual sentiment: Neutral
[2025-03-05 01:18:18,896] INFO __main__:predict_sentiment:247 - Predict: [This was the worst movie I've ever seen. A complete waste of time.                                  ] :        Sentiment: negative, Score: 1.00. Actual sentiment: negative
[2025-03-05 01:18:18,911] INFO __main__:predict_sentiment:247 - Predict: [Absolutely awful! I regret watching this movie.                                                     ] :        Sentiment: negative, Score: 0.99. Actual sentiment: negative
[2025-03-05 01:18:18,924] INFO __main__:predict_sentiment:247 - Predict: [The plot was weak, and the acting was even worse.                                                   ] :        Sentiment: negative, Score: 0.99. Actual sentiment: negative
[2025-03-05 01:18:18,938] INFO __main__:predict_sentiment:247 - Predict: [It had a few good moments, but overall, it was disappointing.                                       ] :        Sentiment: negative, Score: 0.99. Actual sentiment: negative
[2025-03-05 01:18:18,953] INFO __main__:predict_sentiment:247 - Predict: [Not bad, but I wouldn’t watch it again.                                                             ] :        Sentiment: negative, Score: 0.97. Actual sentiment: negative
[2025-03-05 01:18:18,967] INFO __main__:predict_sentiment:247 - Predict: [Es war der schlimmste Film, den ich je gesehen habe.                                                ] :        Sentiment: neutral, Score: 0.83. Actual sentiment: negative
[2025-03-05 01:18:18,981] INFO __main__:predict_sentiment:247 - Predict: [Der Film war eine große Enttäuschung.                                                               ] :        Sentiment: neutral, Score: 0.83. Actual sentiment: negative
[2025-03-05 01:18:18,992] INFO __main__:predict_sentiment:247 - Predict: [Ein durchschnittlicher Film mit einigen guten Momenten.                                             ] :        Sentiment: neutral, Score: 0.87. Actual sentiment: neutral
[2025-03-05 01:18:19,003] INFO __main__:predict_sentiment:247 - Predict: [Der Film hatte eine interessante Handlung.                                                          ] :        Sentiment: neutral, Score: 0.88. Actual sentiment: positive
[2025-03-05 01:18:19,019] INFO __main__:predict_sentiment:247 - Predict: [Fantastischer Film! Ich habe jede Minute genossen.                                                  ] :        Sentiment: neutral, Score: 0.96. Actual sentiment: positive
[2025-03-05 01:18:19,036] INFO __main__:predict_sentiment:247 - Predict: [Questo film era terribile, non lo consiglio a nessuno.                                              ] :        Sentiment: neutral, Score: 0.64. Actual sentiment: negative
[2025-03-05 01:18:19,049] INFO __main__:predict_sentiment:247 - Predict: [Non mi è piaciuto, la trama era confusa.                                                            ] :        Sentiment: neutral, Score: 0.77. Actual sentiment: negative
[2025-03-05 01:18:19,061] INFO __main__:predict_sentiment:247 - Predict: [Un film discreto, ma niente di speciale.                                                            ] :        Sentiment: neutral, Score: 0.85. Actual sentiment: neutral
[2025-03-05 01:18:19,073] INFO __main__:predict_sentiment:247 - Predict: [Molto bello, una storia ben raccontata.                                                             ] :        Sentiment: neutral, Score: 0.82. Actual sentiment: positive
[2025-03-05 01:18:19,088] INFO __main__:predict_sentiment:247 - Predict: [Capolavoro assoluto! Da vedere e rivedere.                                                          ] :        Sentiment: neutral, Score: 0.84. Actual sentiment: positive
[2025-03-05 01:18:19,100] INFO __main__:predict_sentiment:247 - Predict: [Ce film était terrible, je ne le recommande pas.                                                    ] :        Sentiment: negative, Score: 0.99. Actual sentiment: negative
[2025-03-05 01:18:19,111] INFO __main__:predict_sentiment:247 - Predict: [Un film ennuyeux avec peu de bons moments.                                                          ] :        Sentiment: neutral, Score: 0.86. Actual sentiment: negative
[2025-03-05 01:18:19,126] INFO __main__:predict_sentiment:247 - Predict: [Un film moyen, mais agréable.                                                                       ] :        Sentiment: neutral, Score: 0.85. Actual sentiment: neutral
[2025-03-05 01:18:19,138] INFO __main__:predict_sentiment:247 - Predict: [J'ai aimé l'histoire, c'était intéressant.                                                          ] :        Sentiment: neutral, Score: 0.75. Actual sentiment: positive
[2025-03-05 01:18:19,150] INFO __main__:predict_sentiment:247 - Predict: [Incroyable! Un des meilleurs films que j'ai vus.                                                    ] :        Sentiment: neutral, Score: 0.98. Actual sentiment: positive
[2025-03-05 01:18:19,161] INFO __main__:predict_sentiment:247 - Predict: [Esta película fue horrible, no la recomiendo.                                                       ] :        Sentiment: negative, Score: 0.98. Actual sentiment: negative
[2025-03-05 01:18:19,177] INFO __main__:predict_sentiment:247 - Predict: [Un poco aburrida, pero tuvo algunos momentos buenos.                                                ] :        Sentiment: neutral, Score: 0.93. Actual sentiment: negative
[2025-03-05 01:18:19,189] INFO __main__:predict_sentiment:247 - Predict: [Fue un buen filme, pero nada espectacular.                                                          ] :        Sentiment: negative, Score: 0.76. Actual sentiment: neutral
[2025-03-05 01:18:19,212] INFO __main__:predict_sentiment:247 - Predict: [Gran película, realmente disfruté la trama.                                                         ] :        Sentiment: neutral, Score: 0.69. Actual sentiment: positive
[2025-03-05 01:18:19,225] INFO __main__:predict_sentiment:247 - Predict: [Maravilloso! Lo volvería a ver muchas veces.                                                        ] :        Sentiment: neutral, Score: 0.92. Actual sentiment: positive
[2025-03-05 01:18:19,225] INFO __main__:step_04_predict_sentiment:304 - Accuracy on sample_sentences_2.json: 35.71%
[2025-03-05 01:18:19,227] INFO __main__:<module>:320 - Done...

### CODE REVIEW

**Comment by liangzixuan** [comment]
Use base mode: cardiffnlp/twitter-roberta-base-sentiment-latest
IMDB data subset size: train 4000 test 1000, compared to original train 2000 test 500.

**Comment by liangzixuan** [comment]
Accuracy and f1 around 0.92 after 6 epochs, but not doing well on the sample_sentences_2.json... Wanted to use all IMDB dataset but it would take too long to train

**Comment by liangzixuan** [comment]
Also sample_sentences_2.json has multiple languages (en, de, fr, it, es) but IMDB dataset only has English data. Tried using some multilingual models but not working well. Not sure how to fine tune properly
